servicenow/stackstate_checks/servicenow/servicenow.py

```diff
@@ -174,62 +160,59 @@ def _process_components(self, components, instance_config):
+    def _collect_relation_types(self, instance_info):
         """
         collects relations from CMDB
         :return: dict, raw response from CMDB
         """
+        auth = (instance_info.user, instance_info.password)
+        url = instance_info.url + '/api/now/table/cmdb_rel_type?sysparm_fields=sys_id,parent_descriptor'
+        return self._get_json(url, instance_info.timeout, auth)
 
-        base_url = instance_config.base_url
-        auth = instance_config.auth
-        url = base_url + '/api/now/table/cmdb_rel_type?sysparm_fields=sys_id,parent_descriptor'
-
-        return self._get_json(url, instance_config.timeout, auth)
-
-    def _process_relation_types(self, instance_config):
+    def _process_relation_types(self, instance_info):
         """
         collect available relations from cmdb_rel_ci
-        :return: nothing
         """
         relation_types = {}
-        state = self._collect_relation_types(instance_config)
+        types = self._collect_relation_types(instance_info)
 
-        if "result" in state:
-            for relation in state.get('result', []):
-                id = relation['sys_id']
+        if "result" in types:
+            for relation in types.get('result', []):
+                sys_id = relation['sys_id']
                 parent_descriptor = relation['parent_descriptor']
-                relation_types[id] = parent_descriptor
+                relation_types[sys_id] = parent_descriptor
         return relation_types
 
-    def _collect_relations(self, instance_config, offset):
+    def _collect_relations(self, instance_info, offset):
         """
         collect relations between components from cmdb_rel_ci and publish these in batches.
         """
-        base_url = instance_config.base_url
-        auth = instance_config.auth
-        sys_class_filter = instance_config.sys_class_filter
-        url = base_url + '/api/now/table/cmdb_rel_ci'
-        sys_class_filter_query = self.get_sys_class_relation_filter_query(sys_class_filter)
+        auth = (instance_info.user, instance_info.password)
+        url = instance_info.url + '/api/now/table/cmdb_rel_ci'
+        sys_class_filter_query = self.get_sys_class_relation_filter_query(instance_info.sys_class_filter)
         if sys_class_filter_query:
             url = url + "?{}".format(sys_class_filter_query)
             self.log.debug("URL for relation collection after applying filter:- %s", url)
 
-        return self._get_json_batch(url, offset, instance_config.batch_size, instance_config.timeout, auth)
+        return self._get_json_batch(url, offset, instance_info.batch_size, instance_info.timeout, auth)
 
-    def _process_relations(self, relations, instance_config):
-        relation_types = self._process_relation_types(instance_config)
+    def _process_relations(self, relations, instance_info):
+        """
+        process relations
+        """
+        relation_types = self._process_relation_types(instance_info)
```

Comment:
Same thing here. It can potentially fail if `parent` / `child` and the values aren't present. The lookup for `relation_types[type_sys_id]` can also fail.